### PR TITLE
ceph: set default FsGroupChangePolicy value to 'None'

### PR DESCRIPTION
**Pull request by humblec**
The `None` value Indicates that volumes will be mounted
with no modifications, as the CSI volume driver does not
support these operations. While volumes are provisioned
by the CephFS CSI driver the global permissions are set
on the volume and we dont expect the Fsgroup policy or
check from CO side  to play a role here.

Ref #ceph/ceph-csi/../internal/cephfs/nodeserver.go#L190

```
!csicommon.MountOptionContains(fuseMountOptions, readOnly) {
		// #nosec - allow anyone to write inside the stagingtarget path
		err = os.Chmod(stagingTargetPath, 0o777)
```

The current default value ie `ReadWriteOnceWithFSType` cause
volumes to be examined to determine if volume ownership and permissions
should be modified to match the pod's security policy. Changes could occur
if the fsType is defined and the persistent volume's accessModes contains
ReadWriteOnce.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

File: Documentation/helm-operator.md [hunk 1]
```diff
@@ -106,7 +106,7 @@ The following tables lists the configurable parameters of the rook-operator char
 | `csi.provisionerPriorityClassName`  | PriorityClassName to be set on csi driver provisioner pods.                                                                 | <none>                                                    |
 | `csi.enableOMAPGenerator`           | EnableOMAP generator deploys omap sidecar in CSI provisioner pod, to enable it set it to true                               | `false`                                                   |
 | `csi.rbdFSGroupPolicy`              | Policy for modifying a volume's ownership or permissions when the RBD PVC is being mounted                                  | ReadWriteOnceWithFSType                                   |
-| `csi.cephFSFSGroupPolicy`           | Policy for modifying a volume's ownership or permissions when the CephFS PVC is being mounted                               | ReadWriteOnceWithFSType                                   |
+| `csi.cephFSFSGroupPolicy`           | Policy for modifying a volume's ownership or permissions when the CephFS PVC is being mounted                               | `None`                                  |
```
Comment (Madhu-1) [inline Documentation/helm-operator.md:109]: Is this only for cephfs not for rbd?

File: pkg/operator/ceph/csi/spec.go [hunk 2]
```diff
@@ -612,7 +612,7 @@ func startDrivers(clientset kubernetes.Interface, rookclientset rookclient.Inter
 		if err != nil {
 			// logging a warning and intentionally continuing with the default
 			// log level
-			logger.Warningf("failed to parse CSI_CEPHFS_FSGROUPPOLICY. Defaulting to %q. %v", k8scsi.ReadWriteOnceWithFSTypeFSGroupPolicy, err)
+			logger.Warningf("failed to parse CSI_CEPHFS_FSGROUPPOLICY. Defaulting to %s. %v", "None", err)
```
Comment (Madhu-1) [inline pkg/operator/ceph/csi/spec.go:615]: Use predefined const from k8scsi package

File: Documentation/helm-operator.md (hunk 1 above)
Comment (humblec) [reply Documentation/helm-operator.md:109]: Yeah, atleast in this stage, I see only `cephfs` as the possible candidate here, the reason being the global permission set we do and also due to the fact of `rwx` share type volume. Have to think some more before proposing the change for `rbd` and especially while it support both `block and filemode`.

**Comment by humblec** [comment]
@Madhu-1 @travisn looks like the test failure is spurious. can we retrigger the failed test alone ?

**Review by Madhu-1 — APPROVED** [review]

**Review by leseb — APPROVED** [review]

**Pull request merged**